plugs/message/html/render/poll.js

```diff
@@ -0,0 +1,244 @@
+
+      function Positions ({ pollDoc, avatar, timeago, name, mdRenderer }) {
+        return h('section.PollPositions', [
+          h('h2', ['History']),
+          h('div.positions', map(pollDoc.positions, position => {
+            if (!isPosition(position)) return
+            const {author, timestamp} = position.value
+            // postion, reason, time, avatar, name
+            return h('PollPosition', [
+              h('div.left', [
+                h('div.avatar', avatar(author)),
+                h('div.timestamp', timeago(timestamp))
+              ]),
+              h('div.right', [
+                h('div.summary', [
+                  h('div.name', name(author)),
+                  '-',
+                  h('div.choice', position.choice)
+                ]),
+                h('div.reason', mdRenderer(position.reason || ''))
+              ])
+            ])
+          }))
+        ])
+      }
+
+      function Results ({ pollDoc, avatar }) {
+        return h('section.PollResults', [
+          h('h2', 'Current Results'),
+          h('div.choices', map(pollDoc.results, result => {
+            const count = computed(result.voters, vs => Object.keys(vs).length)
+            return when(count, h('div.choice', [
+              h('div.header', [
+                result.choice,
+                h('span.count', ['(', count, ')'])
+              ]),
+              h('div.positions', Object.keys(result.voters).map(avatar))
+            ]))
+          }))
+        ])
+      }
+
+      function NewPosition ({ choices, currentPosition, onPublish }) {
+        const newPosition = Struct({
+          choice: Value(),
+          reason: Value('')
+        })
+
+        const forceShow = Value(false)
+        forceShow(console.log)
+
+        const className = computed([pollDoc.myPosition, forceShow], (myPosition, force) => {
+          if (force) return '-show'
+          if (myPosition === false) return '-hidden'
+          return !myPosition ? '-show' : '-hidden'
+        })
+
+        return h('section.NewPosition', { className }, [
+          h('div.field -choices', [
+            h('label', 'Choose One'),
+            h('div.inputs', choices.map((choice, index) => {
+              var id = `choice-${index}`
+              return h('div.choice', {'ev-click': ev => { newPosition.choice.set(index) }}, [
+                h('input', { type: 'radio', checked: computed(newPosition.choice, c => c === index), id, name: 'choices' }),
+                h('label', { for: id }, choice)
+              ])
+            })
+            )
+          ]),
+          h('div.field -reason', [
+            h('label', 'Reason'),
+            h('textarea', { 'ev-input': ev => newPosition.reason.set(ev.target.value) }, newPosition.reason)
+          ]),
+          h('div.actions', [
+            h('button.publish.-primary', { 'ev-click': publish }, 'Publish position')
+          ]),
+          h('div.changePosition', { 'ev-click': ev => forceShow.set(true) },
```

Comment:
user feedback: felt weird to be missing a cancel button, what if i don't really want to change my position?

Reply:
nice catch